### ISSUE TITLE
local rank 0 logging random values of loss when used auto_metric_logging is set to true ?

### ISSUE THREAD

**Issue by nsriniva03** [comet-ml/issue-tracking#461]
## Describe the Bug
While logging the loss metric comet_ml logs random values for local rank 0 and the right value of loss for other ranks. Look at the graphs below logged by comet_ml.

For rank 0:

<img width="1120" alt="Screen Shot 2022-02-07 at 8 22 06 AM" src="https://user-images.githubusercontent.com/83607477/152806077-19cdbd07-d6d7-4c1c-b44e-7bbe6445ce23.png">

For rank 1:
<img width="1130" alt="Screen Shot 2022-02-07 at 8 22 50 AM" src="https://user-images.githubusercontent.com/83607477/152806132-f37f5497-2475-4ebf-8aba-10522552b1bf.png">

Loss metric logged on different ranks:

<img width="1136" alt="Screen Shot 2022-02-07 at 8 22 35 AM" src="https://user-images.githubusercontent.com/83607477/152806264-aa890997-a652-4ba2-a910-09bc404af0f7.png">

I noticed that a very large value is logged for rank 0.

## Expected behavior
I expect the loss logged on rank 0 to be very similar to that of rank 1. 

## Where is the issue?
- [x ] Comet Python SDK
- [x ] Comet UI
- [ ] Third Party Integrations (Huggingface, TensorboardX, Pytorch Lighting etc)

**Comment by DN6** [comment]
Hello @nsriniva03. Looking into this. Just to confirm, are you running distributed training with Comet? 

Would it be possible to share the code used to run these experiments? It would help our team have more context around what could be happening.

**Comment by nsriniva03** [comment]
@DN6 , I can't share the code but I could put together a skeletal code that highlights the main steps.

**Comment by DN6** [comment]
Thank you! That would be very helpful. Additionally, are you using data parallel training? If so, is your data being distributed the same way across your training nodes? Or is it being shuffled and batches are randomly assigned to nodes across runs?

**Comment by nsriniva03** [comment]
yes, I am using distributed data parallel for training. I am using the distributed data sampler in pytorch  to handle data distribution across ranks. It loads a subset of the data to each rank that is exclusive to it.

**Comment by DN6** [comment]
Understood. 

Would it be possible to run your experiment again and send the data that is currently being sent to rank 0 to another rank to see if the same behavior occurs in this rank?

**Comment by nsriniva03** [comment]
@DN6 , This will take sometime as currently I have a large experiment running. But I will keep you posted.

**Comment by DN6** [comment]
Got it. Would it be possible to take a look at the data being sent to rank 0 to see if there is anything unusual about it? Also if you can share a skeletal version of your code I can try to replicate the issue.
